### PR TITLE
Replaced the `oak_log` crate with module `logger` in `oak` (#645)

### LINKED ISSUE #645: Merge `oak_log` crate in `oak`
There is not a strong reason for keeping this as a separate crate, it does introduce an extra dependency on `log`, but that's a very lightweight one anyways (it's just a facade).

I think we can simply move everything under `oak::log` in the main SDK crate.

Ref. https://github.com/project-oak/oak/tree/master/sdk/rust/oak_log

@rbehjati perhaps you could look into it after #555 ?

In the meanwhile, if anyone thinks of any reason why we should keep these crates separate, please reply here.

### PR DESCRIPTION
**Pull request by rbehjati**
This change moves the functionality in crate 'oak_log' to module `logger` in crate `oak`.

The only advantage of having the logging functionality in a separate crate is to gain more granularity in specifying dependencies. However, we think logging is popular enough to be included as a module in oak.

Fixes #645 

Checklist

- [x] Pull request affects core Oak functionality (e.g. runtime, SDK, ABI)
  - [ ] I have written tests that cover the code changes.
  - [ ] I have checked that these tests are run by [Cloudbuild](cloudbuild.yaml)
  -  [x] I have updated [documentation](docs/) accordingly.
  - [ ] I have raised an [issue](https://github.com/project-oak/oak/issues) to
        cover any TODOs and/or unfinished work.
- [ ] Pull request includes prototype/experimental work that is under
      construction.

### CODE REVIEW

File: docs/sdk.md [hunk 1]
```diff
@@ -104,9 +104,9 @@ provider of persistent storage.
 The [`oak::proto`](https://project-oak.github.io/oak/doc/oak/proto/index.html)
 module holds auto-generated submodules for dealing with protocol buffers.
 
-## `oak_log` Crate
+## `oak::logger` Crate
```
Comment (tiziano88) [inline docs/sdk.md]: This is now a module, not a crate.

File: sdk/rust/oak/Cargo.toml [hunk 2]
```diff
@@ -9,7 +9,7 @@ license = "Apache-2.0"
 oak_abi = "=0.1.0"
 protobuf = "*"
 fmt = "*"
-log = "*"
+log = { version = "^0.4.0", features = ["std"] }
```
Comment (tiziano88) [inline sdk/rust/oak/Cargo.toml]: Why was this needed? We usually don't pin our deps here, we just rely on Cargo.lock for that.

File: sdk/rust/oak/Cargo.toml [hunk 3]
```diff
@@ -21,6 +21,6 @@ protoc-rust = "*"
 [dev-dependencies]
 assert_matches = "*"
 oak_tests = "=0.1.0"
-serial_test = "*"
-serial_test_derive = "*"
+serial_test = "^0.2"
+serial_test_derive = "^0.2"
```
Comment (tiziano88) [inline sdk/rust/oak/Cargo.toml]: Same comment here.

File: sdk/rust/oak/Cargo.toml (hunk 2 above)
Comment (rbehjati) [reply sdk/rust/oak/Cargo.toml]: This is the dependency that was specified in oak_log. I copied it here as I thought it is needed to avoid incompatibilities. I can change it to "*" if you think it is safe. But I agree, since we track Cargo.lock on git, specifying the version here does not matter.

File: sdk/rust/oak/Cargo.toml (hunk 3 above)
Comment (rbehjati) [reply sdk/rust/oak/Cargo.toml]: Same as above :)

File: sdk/rust/oak/Cargo.toml (hunk 2 above)
Comment (tiziano88) [reply sdk/rust/oak/Cargo.toml]: Yes I think we should just use "*" -- as long as it compiles it should be safe (famous last words...). If it does not compile, please feel free to fix whatever is missing inconsistent, or let me know if you think that's not possible.

File: docs/sdk.md [hunk 4]
```diff
@@ -104,9 +104,9 @@ provider of persistent storage.
 The [`oak::proto`](https://project-oak.github.io/oak/doc/oak/proto/index.html)
 module holds auto-generated submodules for dealing with protocol buffers.
 
-## `oak_log` Crate
+### `oak::logger` Module
 
-The [`oak_log`](https://project-oak.github.io/oak/doc/oak_log/index.html) crate
-is a logging implementation for the Rust
+The [`oak::logger`](https://project-oak.github.io/oak/doc/oak/logger/index.html)
+crate is a logging implementation for the Rust
```
Comment (tiziano88) [inline docs/sdk.md]:
```suggestion
module is a logging implementation for the Rust
```
